packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWSelectList/CWSelectList.tsx

```diff
@@ -0,0 +1,123 @@
+import './CWSelectList.scss';
+import React, { useEffect } from 'react';
+import type { GroupBase, Props } from 'react-select';
+import Select, { components } from 'react-select';
+import { CWIcon } from '../../cw_icons/cw_icon';
+import { getClasses } from '../../helpers';
+import { ComponentType } from '../../types';
+import { MessageRow } from '../CWTextInput/MessageRow';
+import { useFormContext } from 'react-hook-form';
+
+type CustomCWSelectListProps = {
+  label?: string;
+  hookToForm?: boolean;
+};
+
+export const CWSelectList = <
+  Option,
+  IsMulti extends boolean = false,
+  Group extends GroupBase<Option> = GroupBase<Option>
+>(
+  props: Props<Option, IsMulti, Group> & CustomCWSelectListProps
+) => {
+  const formContext = useFormContext();
+  const formFieldContext = props.hookToForm
+    ? formContext.register(props.name)
+    : ({} as any);
+  const formFieldErrorMessage =
+    props.hookToForm &&
+    (formContext?.formState?.errors?.[props.name]?.message as string);
+
+  useEffect(() => {
+    props.hookToForm &&
+      formContext &&
+      props.name &&
+      props.defaultValue &&
+      formContext.setValue(props.name, props.defaultValue);
+  }, [props.hookToForm, props.name, props.defaultValue, formContext]);
+
+  useEffect(() => {
+    props.hookToForm &&
+      formContext &&
+      props.name &&
+      props.value &&
+      formContext.setValue(props.name, props.value);
+  }, [props.hookToForm, props.name, props.value, formContext]);
+
+  return (
+    <div>
```

Comment:
maybe `CWSelectList` class would be good to add here, will be easier to find this component in the DOM

Reply:
i agree, will add shortly

Reply:
[Done](https://github.com/hicommonwealth/commonwealth/pull/5086/commits/718240a20d980f0515f69f15e0eeee8aa2e7f459)